### PR TITLE
Add error check for form fields that are not in props

### PR DESCRIPTION
**Pull request by dvingo**
Was debugging a form component and realized I had a typo in the form-fields set, this will make the mistake an error so it can be caught earlier.

### CODE REVIEW

File: src/main/com/fulcrologic/fulcro/algorithms/form_state.cljc [hunk 1]
```diff
@@ -116,6 +116,9 @@
     (when (and (not has-fields?) queries-for-config?)
       (throw (ex-info (str "Attempt to add form configuration to " (rc/component-name class) ", but it does not declare any fields!")
                {:offending-component class})))
+    (when (and has-fields? (empty? fields))
```
Comment (awkay) [inline src/main/com/fulcrologic/fulcro/algorithms/form_state.cljc:119]: what about subform-keys? They couldbe subforms

Comment (dvingo) [reply src/main/com/fulcrologic/fulcro/algorithms/form_state.cljc:119]: thanks, I'll look into this.

Comment (dvingo) [reply src/main/com/fulcrologic/fulcro/algorithms/form_state.cljc:119]: haven't looked into this yet, might have some time in the next few days though

**Comment by dvingo** [comment]
Hi Tony, got a chance to update this, it is now ready for review.

File: src/test/com/fulcrologic/fulcro/algorithms/form_state_spec.cljc [hunk 2]
```diff
@@ -148,6 +163,10 @@
   (component "error checking"
     (let [data-tree {:id 1 :name "A" :thing {:id 2 :x 42}}]
       (assertions
+        "throws an exception if any declared form fields are missing from the component query"
+        (fs/add-form-config FormMissingFields data-tree) =throws=> #"to .*FormMissingFields. It declares fields but not all of them are in the query!"
+        "does not throw an exception if all fields are present in the query when some are sub-forms"
```
Comment (awkay) [inline src/test/com/fulcrologic/fulcro/algorithms/form_state_spec.cljc:168]: In general I recommend using positive language in test assertions. For example, "Can successfully generate the config for a valid form with subforms".  Saying it doesn't do something opens the door to everything in the universe (conceptually). It doesn't paint my house, wash my car, re-render the screen, launch a rocket, etc...saying what it doesn't do is basically saying nothing at all.  

In the case of exceptions I guess it is a grey area, but I think it is clearer to say that a good case succeeds and a bad case fails.

no need to change anything, I just thought I would pass that on as food for thought.

Comment (dvingo) [reply src/test/com/fulcrologic/fulcro/algorithms/form_state_spec.cljc:168]: Heh, I had the same thought. If `add-form-config` is used with any subform that would verify this works. I think adding this under a positive test for subforms would be the way to go for something like this in the future.

Thanks for pointing this out!

**Pull request merged**